app/controllers/ViewController.php

```diff
@@ -7,8 +7,17 @@ public function indexAction(){
         $this->view->tasks = $tasks;
     }
 
-    public function taskAction(){
-        $id = $this->_namedParameters['id'];
-        $this->view->task = $this->database->getTask($id);
+    public function taskAction()
+    {
+      $id = $this->_namedParameters['id'] ?? null;
+      $task = $this->database->getTask($id);  
+
+      extract([
+        'task' => $task,
+        'sidebarIncluded' => true, //My sidebar was duplicated, with this line I am avoiding it
+    ]);
+
+    include __DIR__ . "/../views/scripts/view/task.phtml";
```

Comment:
This `include` shouldn't be here. The task of including the View within the Controller happens with the routing our framework uses. The logic is as follows: 
For each route, the autoloader function by default retrieves the corresponding Controller and View (if the naming follows the framework convention. In our case, it does. This is why you use `taskAction` naming). 

With `include` here you are importing the view twice. This is not necessary and causes a buggy behavior when you reload the `/view/:id` page, because it first loads the content without the styles and then it loads the styles.

In order to inject the $task array into the view, you should use `$this->view->task` = $task. (`$task = $this->database->getTask($id)`). This way, you can retrieve $this->task['title'] from within the view file (`task.phtml`). 

You should remove the whole extract() and include statements from here because they are redundant.

Reply:
Thanks! I was really stuck on this topic and couldn't get it to work